File: Source/Padutronics.Windows.Win32.Api.CWrapper/D2D1/ID2D1Bitmap.cpp
#include "ID2D1Bitmap.hpp"

void ID2D1Bitmap_GetDpi(ID2D1Bitmap& _this, FLOAT* dpiX, FLOAT* dpiY)
{
    _this.GetDpi(dpiX, dpiY);
}

auto ID2D1Bitmap_GetPixelFormat(ID2D1Bitmap& _this) -> D2D1_PIXEL_FORMAT
{
    return _this.GetPixelFormat();
}

auto ID2D1Bitmap_GetPixelSize(ID2D1Bitmap& _this) -> D2D1_SIZE_U
{
    return _this.GetPixelSize();
}

auto ID2D1Bitmap_GetSize(ID2D1Bitmap& _this) -> D2D1_SIZE_F
{
    return _this.GetSize();
}

File: Source/Padutronics.Windows.Win32.Api/D2D1/ID2D1Bitmap.cs
using Padutronics.Windows.Win32.Api.DCommon;
using System.Runtime.InteropServices;

namespace Padutronics.Windows.Win32.Api.D2D1;

[Guid("a2296057-ea42-4099-983b-539fb6505426")]
public interface ID2D1Bitmap : ID2D1Image
{
    void GetDpi(out float dpiX, out float dpiY);
    D2D1_PIXEL_FORMAT GetPixelFormat();
    D2D_SIZE_U GetPixelSize();
    D2D_SIZE_F GetSize();
}